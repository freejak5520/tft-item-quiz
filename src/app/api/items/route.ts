import items from "@/data/items/10.json";
import { NextResponse } from "next/server";

export async function GET() {
  const encodedData = Buffer.from(JSON.stringify(items.data)).toString(
    "base64"
  );
  return NextResponse.json({ data: encodedData });
}
