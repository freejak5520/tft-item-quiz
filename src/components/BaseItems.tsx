import Item from "./Item";

type Props = {
  items: Item[];
  onClick?: (item: Item) => void;
};

const BaseItems = ({ items, onClick }: Props) => {
  return (
    <div className="flex cursor-pointer flex-wrap justify-center gap-2 py-12">
      {items.map((item: Item) => (
        <Item
          key={item.id}
          onClick={() => {
            onClick && onClick(item);
          }}
          item={item}
          size={70}
        />
      ))}
    </div>
  );
};

export default BaseItems;
