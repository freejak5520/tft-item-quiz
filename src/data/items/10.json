{
  "data": [
    {
      "id": 1,
      "name": "B.F. 대검",
      "image": "TFT_Item_BFSword.png"
    },
    {
      "id": 2,
      "name": "곡궁",
      "image": "TFT_Item_RecurveBow.png"
    },
    {
      "id": 3,
      "name": "쇠사슬 조끼",
      "image": "TFT_Item_ChainVest.png"
    },
    {
      "id": 4,
      "name": "음전자 망토",
      "image": "TFT_Item_NegatronCloak.png"
    },
    {
      "id": 5,
      "name": "쓸데없이 큰 지팡이",
      "image": "TFT_Item_NeedlesslyLargeRod.png"
    },
    {
      "id": 6,
      "name": "여신의 눈물",
      "image": "TFT_Item_TearOfTheGoddess.png"
    },
    {
      "id": 7,
      "name": "거인의 허리띠",
      "image": "TFT_Item_GiantsBelt.png"
    },
    {
      "id": 8,
      "name": "연습용 장갑",
      "image": "TFT_Item_SparringGloves.png"
    },
    {
      "id": 9,
      "name": "뒤집개",
      "image": "TFT_Item_Spatula.png"
    },
    {
      "id": 11,
      "name": "죽음의 검",
      "image": "TFT_Item_Deathblade.png",
      "baseItems": [1, 1]
    },
    {
      "id": 12,
      "name": "거인 학살자",
      "image": "TFT_Item_MadredsBloodrazor.png",
      "baseItems": [1, 2]
    },
    {
      "id": 13,
      "name": "밤의 끝자락",
      "image": "TFT_Item_GuardianAngel.png",
      "baseItems": [1, 3]
    },
    {
      "id": 14,
      "name": "피바라기",
      "image": "TFT_Item_Bloodthirster.png",
      "baseItems": [1, 4]
    },
    {
      "id": 15,
      "name": "마법공학 총검",
      "image": "TFT_Item_HextechGunblade.png",
      "baseItems": [1, 5]
    },
    {
      "id": 16,
      "name": "쇼진의 창",
      "image": "TFT_Item_SpearOfShojin.png",
      "baseItems": [1, 6]
    },
    {
      "id": 17,
      "name": "스테락의 도전",
      "image": "TFT_Item_SteraksGage.png",
      "baseItems": [1, 7]
    },
    {
      "id": 18,
      "name": "무한의 대검",
      "image": "TFT_Item_InfinityEdge.png",
      "baseItems": [1, 8]
    },
    {
      "id": 19,
      "name": "True Damage 상징",
      "image": "TFT10_Item_TrueDamageEmblem.png",
      "baseItems": [1, 9]
    },
    {
      "id": 22,
      "name": "붉은 덩굴정령",
      "image": "TFT_Item_RapidFireCannon.png",
      "baseItems": [2, 2]
    },
    {
      "id": 23,
      "name": "거인의 결의",
      "image": "TFT_Item_TitansResolve.png",
      "baseItems": [2, 3]
    },
    {
      "id": 24,
      "name": "루난의 허리케인",
      "image": "TFT_Item_RunaansHurricane.png",
      "baseItems": [2, 4]
    },
    {
      "id": 25,
      "name": "구인수의 격노검",
      "image": "TFT_Item_GuinsoosRageblade.png",
      "baseItems": [2, 5]
    },
    {
      "id": 26,
      "name": "스태틱의 단검",
      "image": "TFT_Item_StatikkShiv.png",
      "baseItems": [2, 6]
    },
    {
      "id": 27,
      "name": "내셔의 이빨",
      "image": "TFT_Item_Leviathan.png",
      "baseItems": [2, 7]
    },
    {
      "id": 28,
      "name": "최후의 속삭임",
      "image": "TFT_Item_LastWhisper.png",
      "baseItems": [2, 8]
    },
    { "id": 29, "name": "8비트 상징", "image": "", "baseItems": [2, 9] },
    {
      "id": 33,
      "name": "덤불 조끼",
      "image": "TFT_Item_BrambleVest.png",
      "baseItems": [3, 3]
    },
    {
      "id": 34,
      "name": "가고일 돌갑옷",
      "image": "TFT_Item_GargoyleStoneplate.png",
      "baseItems": [3, 4]
    },
    {
      "id": 35,
      "name": "크라운가드",
      "image": "TFT_Item_Crownguard.png",
      "baseItems": [3, 5]
    },
    {
      "id": 36,
      "name": "수호자의 맹세",
      "image": "TFT_Item_FrozenHeart.png",
      "baseItems": [3, 6]
    },
    {
      "id": 37,
      "name": "태양불꽃 망토",
      "image": "TFT_Item_RedBuff.png",
      "baseItems": [3, 7]
    },
    {
      "id": 38,
      "name": "굳건한 심장",
      "image": "TFT_Item_NightHarvester.png",
      "baseItems": [3, 8]
    },
    {
      "id": 39,
      "name": "펜타킬 상징",
      "image": "TFT10_Item_PentakillEmblem.png",
      "baseItems": [3, 9]
    },
    {
      "id": 44,
      "name": "용의 발톱",
      "image": "TFT_Item_DragonsClaw.png",
      "baseItems": [4, 4]
    },
    {
      "id": 45,
      "name": "이온 충격기",
      "image": "TFT_Item_IonicSpark.png",
      "baseItems": [4, 5]
    },
    {
      "id": 46,
      "name": "적응형 투구",
      "image": "TFT_Item_AdaptiveHelm.png",
      "baseItems": [4, 6]
    },
    {
      "id": 47,
      "name": "저녁갑주",
      "image": "TFT_Item_SpectralGauntlet.png",
      "baseItems": [4, 7]
    },
    {
      "id": 48,
      "name": "수은",
      "image": "TFT_Item_Quicksilver.png",
      "baseItems": [4, 8]
    },
    {
      "id": 49,
      "name": "재즈 상징",
      "image": "TFT10_Item_JazzEmblem.png",
      "baseItems": [4, 9]
    },
    {
      "id": 55,
      "name": "라바돈의 죽음모자",
      "image": "TFT_Item_RabadonsDeathcap.png",
      "baseItems": [5, 5]
    },
    {
      "id": 56,
      "name": "대천사의 지팡이",
      "image": "TFT_Item_ArchangelsStaff.png",
      "baseItems": [5, 6]
    },
    {
      "id": 57,
      "name": "모렐로노미콘",
      "image": "TFT_Item_Morellonomicon.png",
      "baseItems": [5, 7]
    },
    {
      "id": 58,
      "name": "보석 건틀릿",
      "image": "TFT_Item_JeweledGauntlet.png",
      "baseItems": [5, 8]
    },
    {
      "id": 59,
      "name": "K/DA 상징",
      "image": "TFT10_Item_KDAEmblem.png",
      "baseItems": [5, 9]
    },
    {
      "id": 66,
      "name": "푸른 파수꾼",
      "image": "TFT_Item_BlueBuff.png",
      "baseItems": [6, 6]
    },
    {
      "id": 67,
      "name": "구원",
      "image": "TFT_Item_Redemption.png",
      "baseItems": [6, 7]
    },
    {
      "id": 68,
      "name": "정의의 손길",
      "image": "TFT_Item_UnstableConcoction.png",
      "baseItems": [6, 8]
    },
    {
      "id": 69,
      "name": "이모코어 상징",
      "image": "TFT10_Item_EmoEmblem.png",
      "baseItems": [6, 9]
    },
    {
      "id": 77,
      "name": "워모그의 갑옷",
      "image": "TFT_Item_WarmogsArmor.png",
      "baseItems": [7, 7]
    },
    {
      "id": 78,
      "name": "방패파괴자",
      "image": "TFT_Item_PowerGauntlet.png",
      "baseItems": [7, 8]
    },
    {
      "id": 79,
      "name": "Heartsteel 상징",
      "image": "TFT10_Item_PBJEmblem.png",
      "baseItems": [7, 9]
    },
    {
      "id": 88,
      "name": "도적의 장갑",
      "image": "TFT_Item_ThiefsGloves.png",
      "baseItems": [8, 8]
    },
    {
      "id": 89,
      "name": "펑크 상징",
      "image": "TFT10_Item_PunkEmblem.png",
      "baseItems": [8, 9]
    },
    {
      "id": 99,
      "name": "전략가의 왕관",
      "image": "TFT_Item_ForceOfNature.png",
      "baseItems": [9, 9]
    }
  ]
}
