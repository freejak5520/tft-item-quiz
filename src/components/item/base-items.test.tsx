import { render } from "@testing-library/react";
import BaseItems from "./base-items";
import items from "@/data/items/11.json";
import items10 from "@/data/items/10.json";

describe("BaseItems", () => {
  it("renders the base items Season 10", () => {
    const baseItems = items10.data.filter((item) => !item.baseItems);

    const { container } = render(<BaseItems items={baseItems} />);

    expect(container).toMatchSnapshot();
  });

  it("renders the base items", () => {
    const baseItems = items.data.filter((item) => !item.baseItems);

    const { container } = render(<BaseItems items={baseItems} />);

    expect(container).toMatchSnapshot();
  });
});
