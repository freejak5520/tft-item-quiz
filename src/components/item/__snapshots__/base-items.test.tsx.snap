// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseItems renders the base items 1`] = `
<div>
  <div
    class="mx-auto grid w-fit cursor-pointer grid-cols-3 gap-6 py-12"
  >
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="B.F. 대검"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_BFSword.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_BFSword.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_BFSword.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="곡궁"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_RecurveBow.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_RecurveBow.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_RecurveBow.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="쇠사슬 조끼"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_ChainVest.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_ChainVest.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_ChainVest.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="음전자 망토"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NegatronCloak.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NegatronCloak.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NegatronCloak.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="쓸데없이 큰 지팡이"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NeedlesslyLargeRod.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NeedlesslyLargeRod.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NeedlesslyLargeRod.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="여신의 눈물"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_TearOfTheGoddess.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_TearOfTheGoddess.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_TearOfTheGoddess.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="거인의 허리띠"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_GiantsBelt.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_GiantsBelt.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_GiantsBelt.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="연습용 장갑"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_SparringGloves.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_SparringGloves.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_SparringGloves.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="뒤집개"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_Spatula.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_Spatula.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_Spatula.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
  </div>
</div>
`;

exports[`BaseItems renders the base items Season 10 1`] = `
<div>
  <div
    class="mx-auto grid w-fit cursor-pointer grid-cols-3 gap-6 py-12"
  >
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="B.F. 대검"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_BFSword.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_BFSword.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_BFSword.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="곡궁"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_RecurveBow.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_RecurveBow.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_RecurveBow.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="쇠사슬 조끼"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_ChainVest.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_ChainVest.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_ChainVest.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="음전자 망토"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NegatronCloak.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NegatronCloak.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NegatronCloak.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="쓸데없이 큰 지팡이"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NeedlesslyLargeRod.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NeedlesslyLargeRod.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_NeedlesslyLargeRod.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="여신의 눈물"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_TearOfTheGoddess.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_TearOfTheGoddess.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_TearOfTheGoddess.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="거인의 허리띠"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_GiantsBelt.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_GiantsBelt.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_GiantsBelt.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="연습용 장갑"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_SparringGloves.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_SparringGloves.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_SparringGloves.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
    <div
      class="relative flex h-fit w-fit flex-col items-center overflow-hidden rounded border-2 border-slate-600 p-0 shadow cursor-pointer"
    >
      <img
        alt="뒤집개"
        data-nimg="1"
        decoding="async"
        height="70"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_Spatula.webp&w=256&q=75"
        srcset="/_next/image?url=%2Fimg%2Fitems%2FTFT_Item_Spatula.webp&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fitems%2FTFT_Item_Spatula.webp&w=256&q=75 2x"
        style="color: transparent;"
        width="70"
      />
    </div>
  </div>
</div>
`;
